authentication/models.py

```diff
@@ -42,47 +45,57 @@ class CustomUser(AbstractBaseUser, PermissionsMixin):
     last_login = None
     company = None
     position = None
+    relation_id = None
     is_authenticated = None
 
+
     USERNAME_FIELD = "email"
     REQUIRED_FIELDS = ["user_id", "password", "first_name", "surname", "phone_number"]
     objects = CustomUserManager()
 
     def __str__(self):
         return f"{self.first_name} {self.surname} {self.email}"
-
+        
     @classmethod
     def get_user(cls, *args, **kwargs):
         return cls.objects.get(**kwargs)
-
+    
+    @classmethod
+    def generate_token(cls, payload, token=None):
+        decoded_token = AccessToken(token)
+        decoded_token.payload = {**decoded_token.payload, **payload}
+        return str(decoded_token)
+    
     @classmethod
     def generate_company_related_token(cls, request):
         try:
-            raw_token: str = request.headers.get('Authorization')
+            raw_token:str = request.headers.get('Authorization')
             access_token = raw_token.split(' ')[1]
-            decoded_token = AccessToken(access_token)
-            decoded_token.payload['company_id'] = request.data['company_id']
-            return str(decoded_token)
+            company_id = request.data['company_id']
+            return cls.generate_token({'company_id': company_id}, token=access_token)
         except TokenError as e:
             raise e
```

Comment:
Directly raising `TokenError` without handling it might not provide a friendly error response. Consider catching the exception and raising a DRF's `APIException` with a user-friendly message instead

Reply:
This method will be relocated in the Token manager and TokenError handling will be improved.